x/distribution/types/proposal.go

```diff
@@ -1,24 +1,27 @@
 package types
 
 import (
-	"fmt"
-	"strings"
-
 	sdk "github.com/enigmampc/cosmos-sdk/types"
 	govtypes "github.com/enigmampc/cosmos-sdk/x/gov/types"
+	"gopkg.in/yaml.v2"
 )
 
 const (
-	// ProposalTypeCommunityPoolSpend defines the type for a CommunityPoolSpendProposal
-	ProposalTypeCommunityPoolSpend = "CommunityPoolSpend"
+	ProposalTypeCommunityPoolSpend   = "CommunityPoolSpend"
+	ProposalTypeSecretFoundationTax  = "SecretFoundationTax"
+	ProposalRouteSecretFoundationTax = "SecretFoundationTax"
 )
 
-// Assert CommunityPoolSpendProposal implements govtypes.Content at compile-time
-var _ govtypes.Content = CommunityPoolSpendProposal{}
+var (
+	_ govtypes.Content = CommunityPoolSpendProposal{}
+	_ govtypes.Content = SecretFoundationTaxProposal{}
+)
 
 func init() {
 	govtypes.RegisterProposalType(ProposalTypeCommunityPoolSpend)
 	govtypes.RegisterProposalTypeCodec(CommunityPoolSpendProposal{}, "cosmos-sdk/CommunityPoolSpendProposal")
+	govtypes.RegisterProposalType(ProposalTypeSecretFoundationTax)
+	govtypes.RegisterProposalTypeCodec(SecretFoundationTaxProposal{}, "cosmos-sdk/SecretFoundationTaxProposal")
```

Comment:
Can we do it inside an `x/distribution` `param-change` proposal instead of a new proposal type?
Is there a particular reason we should prefer it this way?

Reply:
Yes. I'll update.